app/models/concerns/form526_claim_fast_tracking_concern.rb

```diff
@@ -188,18 +192,19 @@ def update_form_with_classification_code(classification_code)
   end
 
   def log_max_cfi_metrics_on_submit
-    ClaimFastTracking::DiagnosticCodesForMetrics::DC.intersection(diagnostic_codes).each do |diagnostic_code|
-      next unless disabilities.any? do |dis|
-        diagnostic_code == dis['diagnosticCode']
-      end
+    user = User.find(user_uuid)
+    max_cfi_enabled = Flipper.enabled?(:disability_526_maximum_rating, user) ? 'on' : 'off'
```

Comment:
I pulled the user-centric (and DC-agnostic) logic out of the loop, to improve readability and reduce extraneous user/flipper calls.